### PR TITLE
Deprecate 'keepends' argument in favor of 'drop'

### LINKED ISSUE #2258: Merge `drop` and `keepends` kwargs
`recvuntil()` uses `drop` kwarg, while `recvline()` and `recvlines()` use `keepends`. `keepends` is a logical negation of `drop`, while both serve the same purpose. I find this unnecessarily confusing.

I would like to propose using the same kwarg for the purpose of those functions.

It's a matter of discussion whether `drop` or `keepends` or maybe a new different keyword is better.

The new keyword can be added while preserving the old behaviour.

### PR DESCRIPTION
**Pull request by MrQubo**
Add 'drop' argument in place of 'keepends' for tubes. 'keepends' still works but raises depreciation warning.

Resolves #2258.

### CODE REVIEW

File: pwnlib/tubes/tube.py [hunk 1]
```diff
@@ -420,13 +461,13 @@ def recvlines(self, numlines=2**20, keepends=False, timeout=default):
                 else:
                     break
 
-        if not keepends:
-            lines = [line.rstrip(self.newline) for line in lines]
+        if drop:
+            lines = [line[:-len(self.newline)] for line in lines]
```
Comment (Arusekk) [inline pwnlib/tubes/tube.py]: I hope something ensures self.newline is non-empty

File: pwnlib/tubes/tube.py [hunk 2]
```diff
@@ -520,20 +561,23 @@ def recvline(self, keepends=True, timeout=default):
             b'trailing data'
             >>> t.recvline() # doctest: +ELLIPSIS
```
Comment (Arusekk) [inline pwnlib/tubes/tube.py]: Should also work w/o ELLIPSIS

File: pwnlib/tubes/tube.py [hunk 3]
```diff
@@ -99,7 +137,7 @@ def recv(self, numb = None, timeout = default):
             True
             >>> with context.local(log_level='debug'):
             ...    _ = t.recv() # doctest: +ELLIPSIS
-            [...] Received 0xc bytes:
+            [DEBUG] Received 0xc bytes:
```
Comment (Arusekk) [inline pwnlib/tubes/tube.py]: Can probably break if NOTERM is not set; not sure tho

File: pwnlib/tubes/tube.py [hunk 4]
```diff
@@ -44,6 +44,44 @@ def __init__(self, timeout = default, level = None, *a, **kw):
         self._newline = None
         atexit.register(self.close)
 
+    def _normalize_drop_keepends(self, drop, keepends, drop_default):
```
Comment (Arusekk) [inline pwnlib/tubes/tube.py]: Nice.

**Review by Arusekk — CHANGES_REQUESTED** [review]

File: pwnlib/tubes/tube.py (hunk 1 above)
Comment (MrQubo) [reply pwnlib/tubes/tube.py]: That's copy pasted from https://github.com/Gallopsled/pwntools/blob/f2f55f3e1c26e41e7b23f2fab0ff41baf7ebb27c/pwnlib/tubes/tube.py#L576
I think it's better than `.rstrip()`.

File: pwnlib/tubes/tube.py (hunk 3 above)
Comment (MrQubo) [reply pwnlib/tubes/tube.py]: It's with `DEBUG` here: https://github.com/Gallopsled/pwntools/blob/f2f55f3e1c26e41e7b23f2fab0ff41baf7ebb27c/pwnlib/tubes/tube.py#L1066

File: pwnlib/tubes/tube.py (hunk 2 above)
Comment (MrQubo) [reply pwnlib/tubes/tube.py]: Worked for me locally as well, but most (all?) tests in this file have this. I'll remove it everywhere and we'll check if it passes.

**Comment by MrQubo** [comment]
I have like 30 emails from failing pipeline in my inbox. xd

File: pwnlib/tubes/tube.py (hunk 3 above)
Comment (Arusekk) [reply pwnlib/tubes/tube.py]: Sure but what if it is `'[\33[91mDEBUG\33[m] Something'`? I think this was the reason.

Comment (MrQubo) [reply pwnlib/tubes/tube.py]: Ok, I also changed a few other tests I found throughout the codebase to use `[...]`.

File: pwnlib/tubes/tube.py (hunk 1 above)
Comment (MrQubo) [reply pwnlib/tubes/tube.py]: > I hope something ensures self.newline is non-empty

Nothing checks if it's non-empty neither in tubes nor in context. Maybe it should though?

**Comment by peace-maker** [comment]
When inserting the `drop` argument before the `keepends` one we'll break code using positional arguments like `recvline(False)`. Can you switch it up so drop is added later?

It's annoying but we can switch the order when doing a major version bump. What do you think @Arusekk ?

**Comment by Arusekk** [comment]
@peace-maker I agree we should start working on Pwntools 5 soon, dropping py2, adding type annotations, and cleaning up dependencies, inconsistent interfaces and useless defaults.

**Comment by MrQubo** [comment]
@peace-maker I was thing of just removing keepends altogether.

File: pwnlib/tubes/tube.py (hunk 1 above)
Comment (MrQubo) [reply pwnlib/tubes/tube.py]: Actually, recvline can return line without newline, so it should be rstrip.

https://github.com/Gallopsled/pwntools/blob/6f0793eebcf78eab4ac86bb8fcbb1ffee3e59fa6/pwnlib/tubes/tube.py#L531-L534

And that other line of code I linked two comments above is also wrong.

**Comment by MrQubo** [comment]
> When inserting the `drop` argument before the `keepends` one we'll break code using positional arguments like `recvline(False)`. Can you switch it up so drop is added later?
> 
> It's annoying but we can switch the order when doing a major version bump. What do you think @Arusekk ?

That was resolved.
Are we merging this?

**Review by peace-maker — APPROVED** [review]

**Pull request merged**